pkg/client/fake/client.go

```diff
@@ -350,6 +384,20 @@ func (t versionedTracker) Update(gvr schema.GroupVersionResource, obj runtime.Ob
 		return err
 	}
 
+	if t.withStatusSubresource.Has(gvk) {
+		if isStatus { // copy everything but status and metadata.ResourceVersion from original object
+			if err := copyNonStatusFrom(oldObject, obj); err != nil {
+				return fmt.Errorf("failed to copy non-status field for object with status subresouce: %w", err)
+			}
+		} else { // copy status from original object
+			if err := copyStatusFrom(oldObject, obj); err != nil {
+				return fmt.Errorf("failed to copy the status for object with status subresource: %w", err)
+			}
+		}
+	} else if isStatus {
+		return apierrors.NewNotFound(gvr.GroupResource(), accessor.GetName())
```

Comment:
@alvaroaleman just fyi

Regarding:
> Any test that relied on the previous behavior would pass incorrectly though, as it
didn't match what the Kubernetes API does.

In my experience this change is relatively safe as all Update/Patch calls will fail with this error hard

Reply:
Ah that is assuming that folks where using the status client correctly. Of course doesn't apply if they used the regular Patch/Update calls